### ISSUE TITLE
union has to be within a service

### ISSUE THREAD

**Issue by justin2004** [SPARQL-Anything/sparql.anything#173]
This query
```
curl --silent 'http://localhost:3000/sparql.anything'  \
-H 'Accept: text/csv' \
--data-urlencode 'query=
PREFIX xyz: <http://sparql.xyz/facade-x/data/>
PREFIX fx: <http://sparql.xyz/facade-x/ns/>
select *
WHERE {
service <x-sparql-anything:> {
fx:properties fx:location "https://sparql-anything.cc/example1.json" .
{
    [] xyz:genres [ ?p ?o ]
}
union
{
    [] xyz:stars [ ?p ?o ]
}
}
}'
```

yields:
```
p,o
http://www.w3.org/1999/02/22-rdf-syntax-ns#_1,Comedy
http://www.w3.org/1999/02/22-rdf-syntax-ns#_2,Romance
http://www.w3.org/1999/02/22-rdf-syntax-ns#_1,Comedy
http://www.w3.org/1999/02/22-rdf-syntax-ns#_2,Romance
http://www.w3.org/1999/02/22-rdf-syntax-ns#_1,Jennifer Aniston
http://www.w3.org/1999/02/22-rdf-syntax-ns#_2,Courteney Cox
...
```

as expected.

But this query:
```
curl --silent 'http://localhost:3000/sparql.anything'  \
-H 'Accept: text/csv' \
--data-urlencode 'query=
PREFIX xyz: <http://sparql.xyz/facade-x/data/>
PREFIX fx: <http://sparql.xyz/facade-x/ns/>
select *
WHERE {
fx:properties fx:location "https://sparql-anything.cc/example1.json" .
{
    [] xyz:genres [ ?p ?o ]
}
union
{
    [] xyz:stars [ ?p ?o ]
}
}'
```

yields no solutions. Shouldn't they find the same solutions?

**Comment by luigi-asprino** [comment]
maybe 0cb6951 wasn't needed since we addressed the problem with #175, but hey why not having another unit test :-)

**Comment by enridaga** [comment]
I think we agreed that fx properties need to be inside `SERVICE` so this on is won't fix
